### PR TITLE
Feat/Submodel registration and production SAMM

### PR DESCRIPTION
**Pull request by ReneSchroederLJ**
## Description
- refactored the registration of submodels to be reusable
- refactored the contract definitions to be reusable
- implemented the registration of submodels for production, demand and delivery
- implemented SAMM models for Planned Production Output

## Pre-review checks

Please ensure to do as many of the following checks as possible, before asking for committer review:

- [x] DEPENDENCIES are up-to-date. [Dash license tool](https://github.com/eclipse/dash-licenses). Committers can open IP issues for restricted libs.
- [x] [Copyright and license header](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-02) are present on all affected files

### CODE REVIEW

File: backend/src/main/java/org/eclipse/tractusx/puris/backend/common/edc/domain/model/SubmodelType.java [hunk 1]
```diff
@@ -23,6 +23,9 @@
 public enum SubmodelType {
     DTR("none", "none"),
     ITEM_STOCK("urn:samm:io.catenax.item_stock:2.0.0#ItemStock", "$value"),
+    PRODUCTION("urn:samm:io.catenax.planned_production_output:2.0.0 ", "$value"),
+    DEMAND("urn:samm:io.catenax.short_term_material_demand:2.0.0 ", "$value"),
```
Comment (tom-rm-meyer-ISST) [inline backend/src/main/java/org/eclipse/tractusx/puris/backend/common/edc/domain/model/SubmodelType.java:27]: please adapt demand to version 1.0.0

File: backend/src/main/java/org/eclipse/tractusx/puris/backend/common/edc/domain/model/SubmodelType.java [hunk 2]
```diff
@@ -23,6 +23,9 @@
 public enum SubmodelType {
     DTR("none", "none"),
     ITEM_STOCK("urn:samm:io.catenax.item_stock:2.0.0#ItemStock", "$value"),
+    PRODUCTION("urn:samm:io.catenax.planned_production_output:2.0.0 ", "$value"),
```
Comment (tom-rm-meyer-ISST) [inline backend/src/main/java/org/eclipse/tractusx/puris/backend/common/edc/domain/model/SubmodelType.java:26]: Please add Entity name after urns with "#EntityName"
Sidenote: the models can be found in the following repo: https://github.com/eclipse-tractusx/sldt-semantic-models

File: backend/src/main/java/org/eclipse/tractusx/puris/backend/common/edc/logic/service/EdcAdapterService.java [hunk 3]
```diff
@@ -134,7 +132,30 @@ public boolean registerAssetsInitially() {
...
+            variablesService.getItemStockSubmodelEndpoint(),
+            SubmodelType.ITEM_STOCK.URN_SEMANTIC_ID
+        )));
+        log.info("Registration of Planned Production 2.0.0 submodel successful {}", (assetRegistration = registerSubmodelAsset(
+            "Planned Production",
+            variablesService.getProductionSubmodelApiAssetId(),
+            variablesService.getProductionSubmodelEndpoint(),
+            SubmodelType.PRODUCTION.URN_SEMANTIC_ID
+        )));
+        log.info("Registration of Short Term Material Demand 2.0.0 submodel successful {}", (assetRegistration = registerSubmodelAsset(
```
Comment (tom-rm-meyer-ISST) [inline backend/src/main/java/org/eclipse/tractusx/puris/backend/common/edc/logic/service/EdcAdapterService.java:147]: Please update version for demand in log

File: backend/src/main/java/org/eclipse/tractusx/puris/backend/common/edc/logic/service/EdcAdapterService.java [hunk 4]
```diff
@@ -151,30 +172,32 @@ public boolean registerAssetsInitially() {
      */
     public boolean createPolicyAndContractDefForPartner(Partner partner) {
         boolean result = createPolicyDefinitionForPartner(partner);
-        result &= createItemStockSubmodelContractDefinitionForPartner(partner);
+        result &= createSubmodelContractDefinitionForPartner("ItemStock", variablesService.getItemStockSubmodelApiAssetId(), partner);
```
Comment (tom-rm-meyer-ISST) [inline backend/src/main/java/org/eclipse/tractusx/puris/backend/common/edc/logic/service/EdcAdapterService.java:175]: As the other parameters use a black, please also use it for Stock. 

Should we maybe just log the semanticId and use the SubmodelType? Actually, I don't have a preference here

File: backend/src/main/java/org/eclipse/tractusx/puris/backend/common/edc/logic/util/EdcRequestBodyBuilder.java [hunk 5]
```diff
@@ -315,28 +317,27 @@ public JsonNode buildDtrRegistrationBody() {
...
         var propertiesObject = MAPPER.createObjectNode();
         body.set("properties", propertiesObject);
         var dctTypeObject = MAPPER.createObjectNode();
         propertiesObject.set("dct:type", dctTypeObject);
         dctTypeObject.put("@id", "cx-taxo:Submodel");
         propertiesObject.put("cx-common:version", "3.0");
-        var semanticId = MAPPER.createObjectNode();
-        propertiesObject.set("aas-semantics:semanticId", semanticId);
-        semanticId.put("@id", "urn:samm:io.catenax.item_stock:2.0.0#ItemStock");
+        var semanticIdPart = MAPPER.createObjectNode();
```
Comment (tom-rm-meyer-ISST) [inline backend/src/main/java/org/eclipse/tractusx/puris/backend/common/edc/logic/util/EdcRequestBodyBuilder.java:329]: Why renamed to `Part`? Else I would go for `Object` just for unity reasons.

File: backend/src/main/java/org/eclipse/tractusx/puris/backend/common/edc/domain/model/SubmodelType.java (hunk 2 above)
Comment (tom-rm-meyer-ISST) [reply backend/src/main/java/org/eclipse/tractusx/puris/backend/common/edc/domain/model/SubmodelType.java:26]: sidenote: Please remove the spaces at the end of the type.

**Review by tom-rm-meyer-ISST — CHANGES_REQUESTED** [review]
Good refactoring! A few changes needed, some questions just for curiosity.

File: backend/src/main/java/org/eclipse/tractusx/puris/backend/common/edc/logic/service/EdcAdapterService.java (hunk 4 above)
Comment (ReneSchroederLJ) [reply backend/src/main/java/org/eclipse/tractusx/puris/backend/common/edc/logic/service/EdcAdapterService.java:175]: Using the asset id actually makes sense. There's no need to add an extra parameter just for logging purposes.

**Review by tom-rm-meyer-ISST — APPROVED** [review]
LGTM, thanks a lot!

**Pull request merged**